django_lightweight_queue/job.py

```diff
@@ -59,7 +62,13 @@ def run(self):
                 instance.process_job(self)
 
         try:
-            result = self.get_fn().fn(*self.args, **self.kwargs)
+            task = self.get_task_instance()
+
+            if task.atomic:
+                with transaction.atomic():
+                    result = task.fn(*self.args, **self.kwargs)
+            else:
+                result = task.fn(*self.args, **self.kwargs)
```

Comment:
This might be nicer setting a dummy context manager, then using like:
```
atomic_or_noop = transaction.atomic if task.atomic else dummy
with atomic_or_noop():
  blah
```
to avoid repeating oneself?

Reply:
I considered this but decided against it because:

1. It takes more lines of code in total (due to having to define `dummy`),
2. I was worried it makes it slightly less clear what's going on: it looks like there's a context manager that gets run in every case, one has to go to the definition of dummy (or at least infer from the name) that it has no action.

Also thought about factoring out the call, either by doing:

```python
def run():
    nonlocal result
    result = task.fn(*self.args, **self.kwargs)

if task.atomic:
    with transaction.atomic():
        run()
else:
    run()
```

or:

```python
run = functools.partial(task.fn, *self.args, **self.kwargs)

if task.atomic:
    with transaction.atomic():
        result = run()
else:
    result = run()
```

Either is fine but I felt like the original spelling was a little clearer, even though it unfortunately means some duplication.

Open to changing this if you have another way of spelling it though!

Reply:
I don't feel hugely strongly here, but using a tiny portable implementation like:
```
@contextlib.contextmanager
def noop():
    yield
```
and using the names/structure from my previous snippet seem like they'd be pretty clear. Moreso than either of the examples you gave.

PS: https://bugs.python.org/issue10049 lol 2010, and lol that it's _exactly_ the same issue :).